src/parser_fill_cmd.c

```diff
@@ -0,0 +1,123 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   parser_fill_cmd.c                                  :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: sqiu <sqiu@student.42vienna.com>           +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/07/22 14:59:22 by sqiu              #+#    #+#             */
+/*   Updated: 2023/07/28 14:51:22 by sqiu             ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+/**
+ * @file parser_fill_cmd.c
+ * @brief Functions to fill the cmd struct.
+ */
+
+#include "mod_parser.h"
+
+/**
+ * @brief Open file and save fd as fd_in.
+ * 
+ * @param lst 		List of token.
+ * @param new 		New cmd to be filled.
+ * @return t_err 	ERR_BAD_FD, ERR_CLOSE, SUCCESS
+ */
+t_err	ft_save_infile(t_tkn_list **lst, t_cmd *new)
+{
+	int			fd_in;
+	t_tkn_list	*tmp;
+
+	tmp = *lst;
+	tmp = tmp->next;
+	new->infile = 1;
+	if (new->fd_in > -1)
```

Comment:
I think this if clause wants to be below the close() function

Reply:
Why? It is checking whether there already has been an infile opened previously (as new->fd_in is initialised as -1) and in that case it would close it. If not, it will jump over close and proceed to opening the file.

Reply:
True! I confusion!